### ISSUE TITLE
[ColumnSelection] allow to append rows dynamically simple

### ISSUE THREAD

**Issue by SimonHeimberg** [EmchBerger/cube-common-bundle#2]
Modifying dynamically a table with hidden columns should work automatically or with simple helpers.

maybe
 * provide a convenient function to filter the new rows (before inserting)
 * do hiding automatically
   * add a change listener to the table (slow)
   * set column to visible collapse, is not supported by many browsers
   * set style in stylesheet (with `stylesheet.addRule()`, `rule.selectorText = xxx`, ...)
    set class to each hidable column optionally (when colspan is used)
    https://davidwalsh.name/add-rules-stylesheets
 * do column filtering on server side (how??? hiding in twig (macro in filterHelper), read current state in Controller??? )

**Comment by SimonHeimberg** [comment]
```js
ruleNode = $('#columnsTableXxx');
if (ruleNode) {
     sheet = ruleNode[0].sheet;
} else {
    style = document.createElement('style');
    style.style.appendChild(document.createTextNode("")); //webkit hack
    style.setAttr('id', 'columnsTableXxx');
    document.head.appendChild(style);
    sheet = style.sheet;
}
rule = cssRules.item(columnSelector);
if (rule) {
    rule.style.display = 'visible';
} else {
    cssRules.insertRule(columnSelector+'{'+'display: '+'visible'+'}', 0);
}

**Comment by SimonHeimberg** [comment]
@MartinRosserEB what about the changeset shown above?
Because the hiding is saved in a rule, it is automatically applied to new cells.

Used in one place currently: http://hesi.dev.cubetools.ch/tm/app_dev.php/execution/testrun/
Your login is the same as on the productive system.

**Comment by SimonHeimberg** [comment]
tested (with MartinRosserEB) on:
* FireFox
* Edge
* IE
* Chrome

**Comment by SimonHeimberg** [comment]
Can be further improved by doing filtering in twig. But this need some more macros, which must be applied everywhere.
pro:
* transmit less data
contra:
* more effort to apply everywhere

**Comment by SimonHeimberg** [comment]
hiding currently (1.1.1) done with display none+adapting of total column count

    shortcoming:
    * works only for colspan of full length

Tries with "visibility: collapse" on colgroup>col + hiding cells failed (1.1.0)

TODO:

* [ ] implement correctly (for all colspan), but working

to consider
 * browsers IE, Firefox, Chrome, Edge
 * table-layout: fixed (../doc/document, ../meetings/action) / auto (../reservation)

**Comment by MartinRosserEB** [comment]
I suggest to only support table-layout: auto;

We have issues with table-layout: fixed since content must overflow, and we cannot set a width to all columns.

**Comment by SimonHeimberg** [comment]
hopefully fixed with pr #61

**Comment by SimonHeimberg** [comment]
last issue in #64
